engine/graphics/src/opengl/graphics_opengl.cpp

```diff
@@ -3169,33 +3181,37 @@ static void LogFrameBufferError(GLenum status)
             case TEXTURE_FORMAT_R_BC4:
             case TEXTURE_FORMAT_RG_BC5:
             case TEXTURE_FORMAT_RGBA_BC7:
-                if (params.m_DataSize > 0) {
-                    if (tex->m_Type == TEXTURE_TYPE_2D) {
-                        if (params.m_SubUpdate) {
+                if (params.m_DataSize > 0)
+                {
+                    if (tex->m_Type == TEXTURE_TYPE_2D)
+                    {
+                        if (params.m_SubUpdate)
+                        {
                             glCompressedTexSubImage2D(GL_TEXTURE_2D, params.m_MipMap, params.m_X, params.m_Y, params.m_Width, params.m_Height, gl_format, params.m_DataSize, params.m_Data);
-                        } else {
+                        }
+                        else
+                        {
                             glCompressedTexImage2D(GL_TEXTURE_2D, params.m_MipMap, gl_format, params.m_Width, params.m_Height, 0, params.m_DataSize, params.m_Data);
                         }
                         CHECK_GL_ERROR;
-                    } else if (tex->m_Type == TEXTURE_TYPE_2D_ARRAY) {
-                    #ifdef __ANDROID__
-                        #define COMPRESSED_TEX_SUB_IMAGE_3D PFN_glCompressedTexSubImage3D
-                        #define COMPRESSED_TEX_IMAGE_3D     PFN_glCompressedTexImage3D
-                    #else
-                        #define COMPRESSED_TEX_SUB_IMAGE_3D glCompressedTexSubImage3D
-                        #define COMPRESSED_TEX_IMAGE_3D     glCompressedTexImage3D
-                    #endif
-                        if (params.m_SubUpdate) {
-                            COMPRESSED_TEX_SUB_IMAGE_3D(GL_TEXTURE_2D_ARRAY, params.m_MipMap, params.m_X, params.m_Y, params.m_Z, params.m_Width, params.m_Height, params.m_Depth, gl_format, gl_type, params.m_Data);
-                        } else {
-                            COMPRESSED_TEX_IMAGE_3D(GL_TEXTURE_2D_ARRAY, params.m_MipMap, gl_format, params.m_Width, params.m_Height, params.m_Depth, 0, params.m_DataSize, params.m_Data);
+                    }
+                    else if (tex->m_Type == TEXTURE_TYPE_2D_ARRAY)
+                    {
+                        if (params.m_SubUpdate)
+                        {
+                            DMGRAPHICS_COMPRESSED_TEX_SUB_IMAGE_3D(GL_TEXTURE_2D_ARRAY, params.m_MipMap, params.m_X, params.m_Y, params.m_Z, params.m_Width, params.m_Height, params.m_Depth, gl_format, gl_type, params.m_Data);
+                        }
+                        else
+                        {
+                            DMGRAPHICS_COMPRESSED_TEX_IMAGE_3D(GL_TEXTURE_2D_ARRAY, params.m_MipMap, gl_format, params.m_Width, params.m_Height, params.m_Depth, 0, params.m_DataSize * params.m_Depth, params.m_Data);
```

Comment:
This is the actual issue that is fixed - when we create the texture we need to pass in the data size of the ENTIRE texture with all the slices and so forth, but the params struct only contains the data size for ONE slice.